pkg/template/admission/impersonation/admission.go

```diff
@@ -0,0 +1,68 @@
+package impersonation
+
+import (
+	"errors"
+	"io"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/client"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	userapi "github.com/openshift/origin/pkg/user/api"
+	"k8s.io/kubernetes/pkg/admission"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+func init() {
+	admission.RegisterPlugin("openshift.io/TemplateImpersonationAdmissionPlugin", func(c clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return &impersonationAdmissionHandler{
+			Handler: admission.NewHandler(admission.Create, admission.Update),
+		}, nil
+	})
+}
+
+type impersonationAdmissionHandler struct {
+	*admission.Handler
+	oc client.Interface
+}
+
+func (iah *impersonationAdmissionHandler) Admit(attr admission.Attributes) (err error) {
+	templateInstance, ok := attr.GetObject().(*templateapi.TemplateInstance)
+	if !ok {
+		return nil
+	}
+
+	if attr.GetOperation() == admission.Update {
+		oldTemplateInstance := attr.GetOldObject().(*templateapi.TemplateInstance)
+		if templateInstance.Spec.ImpersonateUser != oldTemplateInstance.Spec.ImpersonateUser {
+			return admission.NewForbidden(attr, errors.New("spec.impersonateUser is immutable"))
+		}
+	}
```

Comment:
is there actually a good reason to prevent this, as opposed to using the same logic as is used below to handle the create case?  (if i can create an instance with a particular impersonation value, is there any point in preventing me from updating an instance to have that same impersonation value?)